builtin/providers/azurerm/encryption_settings.go

```diff
@@ -0,0 +1,163 @@
+package azurerm
+
+import (
+	"github.com/Azure/azure-sdk-for-go/arm/compute"
+	"github.com/Azure/azure-sdk-for-go/arm/disk"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func encryptionSettingsSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		MaxItems: 1,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"enabled": {
+					Type:     schema.TypeBool,
+					Required: true,
+
+					// Azure can change enabled from false to true, but not the other way around, so
+					//   to keep idempotency, we'll conservatively set this to ForceNew=true
+					// TODO: Is this the right behavior?
+					ForceNew: true,
+				},
+
+				"disk_encryption_key": {
+					Type:     schema.TypeList,
```

Comment:
@tombuildsstuff, @whiskeyjay

I'm not sure that using a set over a list is worth it for consistency reasons, given all its disadvantages (see https://github.com/hashicorp/terraform/pull/14261).  Is there any other reason not to go with a list?  Thx.

I also notice that azurerm_redis_cache is using a list for this purpose (redis_configuration).